### DISCUSSION TITLE
How to enable highlighting feature but still prevent text copy for LCP protected EPUBs?

### DISCUSSION THREAD

**Discussion by nareshsvn** [readium/kotlin-toolkit#88] [Q&A]
Hi

Currently, I am using the following code to enable the text select/highlight(annotations, highlighting) feature for LCP protected publication. But this code allows copying the publication.
Is there any way that, we can enable the text select/highlight(annotations, highlighting) feature without the text copying feature? Please suggest.

server.loadCustomResource("""
        * {
            user-select: auto !important;
            -webkit-user-select: auto !important;
        }
    """.byteInputStream(), "fix-select.css", Injectable.Style)   //#414

Thanks,
Naresh

**Comment by mickael-menu** [comment]
You need to create a custom `selectionActionModeCallback` for EPUB, and create your own selection menu without the Copy option.

* https://github.com/readium/kotlin-toolkit/blob/adeb8f452fbd48b6ab7e1132264bcce9ab64ee95/test-app/src/main/java/org/readium/r2/testapp/reader/EpubReaderFragment.kt#L91-L95
* https://github.com/readium/kotlin-toolkit/blob/adeb8f452fbd48b6ab7e1132264bcce9ab64ee95/test-app/src/main/java/org/readium/r2/testapp/reader/BaseReaderFragment.kt#L166-L190
* https://github.com/readium/kotlin-toolkit/blob/adeb8f452fbd48b6ab7e1132264bcce9ab64ee95/test-app/src/main/res/menu/menu_reader.xml

However, this will not work when using a physical keyboard as the user can do Ctrl-C directly.

I didn't look further into addressing this issue, and maybe other edge cases. If you have some ideas feel free to share.

**Reply by nareshsvn** [reply]
Thank you for your reply.

In the latest SDK I am able to use the text highlighting feature without a copy option. 
1. In case the user wants to enable the copy feature along with the text highlighting feature in the latest SDK.  is there a way to do it? Please suggest.

Thanks,
Naresh
